src/components/ConfirmationModal.tsx

```diff
@@ -51,7 +51,7 @@ const ConfirmationModal = (props: ConfirmationModalProps): JSX.Element => {
   const closeModal = (): void => setVisible(false)
 
   return (
-    <Modal testID='modal' visible={visible} transparent animationType='fade'>
+    <Modal testID='modal' visible={visible} transparent animationType='fade' onRequestClose={() => setVisible(false)}>
```

Comment:
Not sure if this is really the wanted behaviour. When now being in an exercise and click back-button, the modal opens, clicking back-button again the modal closes. Shouldn't the back-button then confirm the back-navigation?

Reply:
I think this is fine. Its the same behavior as other apps with confirmation dialogs.